hr_timesheet_purchase_order/models/hr_employee_base.py

```diff
@@ -8,3 +12,237 @@ class HrEmployeeBase(models.AbstractModel):
+            ("first", "First"),
+            ("second", "Second"),
+            ("third", "Third"),
+            ("last", "Last"),
+        ],
+        default="first",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_weekday = fields.Selection(
+        [
+            ("mon", "Monday"),
+            ("tue", "Tuesday"),
+            ("wed", "Wednesday"),
+            ("thu", "Thursday"),
+            ("fri", "Friday"),
+            ("sat", "Saturday"),
+            ("sun", "Sunday"),
+        ],
+        string="Day Of The Week",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_month = fields.Selection(
+        [
+            ("january", "January"),
+            ("february", "February"),
+            ("march", "March"),
+            ("april", "April"),
+            ("may", "May"),
+            ("june", "June"),
+            ("july", "July"),
+            ("august", "August"),
+            ("september", "September"),
+            ("october", "October"),
+            ("november", "November"),
+            ("december", "December"),
+        ],
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    repeat_show_dow = fields.Boolean(compute="_compute_repeat_visibility")
+    repeat_show_day = fields.Boolean(compute="_compute_repeat_visibility")
+    repeat_show_week = fields.Boolean(compute="_compute_repeat_visibility")
+    repeat_show_month = fields.Boolean(compute="_compute_repeat_visibility")
+
+    @api.depends(
+        "is_auto_po_generate", "repeat_unit", "repeat_on_month", "repeat_on_year"
+    )
+    def _compute_repeat_visibility(self):
+        """Based on the selected parameters sets
+        the fields that should be visible to the user
+        """
+        for item in self:
+            item.repeat_show_day = (
+                item.is_auto_po_generate
+                and (item.repeat_unit == "month" and item.repeat_on_month == "date")
+                or (item.repeat_unit == "year" and item.repeat_on_year == "date")
+            )
+            item.repeat_show_week = (
+                item.is_auto_po_generate
+                and (item.repeat_unit == "month" and item.repeat_on_month == "day")
+                or (item.repeat_unit == "year" and item.repeat_on_year == "day")
+            )
+            item.repeat_show_dow = (
+                item.is_auto_po_generate and item.repeat_unit == "week"
+            )
+            item.repeat_show_month = (
+                item.is_auto_po_generate and item.repeat_unit == "year"
+            )
+
+    @api.depends("is_auto_po_generate")
+    def _compute_repeat(self):
+        rec_fields = self._get_recurrence_fields()
+        defaults = self.default_get(rec_fields)
+        for employee in self:
+            for f in rec_fields:
+                if employee.recurrence_id:
+                    employee[f] = employee.recurrence_id[f]
+                else:
+                    if employee.is_auto_po_generate:
```

Comment:
This condition you coulf write in one line:
```employee[f] = default.get(f) if employee.is_auto_po_generate else False```

Reply:
Thanks, I'll fix it